main-serve-simple.py

```diff
@@ -17,34 +15,35 @@
 
 
 @ray.remote
-def log_frame(df: pd.DataFrame) -> DatasetProfile:
-    profile = DatasetProfile("")
-    profile.track_dataframe(df)
-    return profile
+def log_frame(df: pd.DataFrame) -> DatasetProfileView:
+    profile = DatasetProfile()
+    profile.track(df)
+    return profile.view()
 
 
 @serve.deployment()
 class Logger:
     def __init__(self) -> None:
-        # TODO how long will this local state stick around?
-        # TODO How many instances of Logger are actually "deployed"?
-        # TODO Can I pull state for all instances of Logger at some point to retrieve these profiles or do I need to coordinate that with some supervisor actor?
-        self.profile = DatasetProfile("")
+        # TODO having to make views like this is weird when the DatasetProfileView constructor exists
+        self.profile = DatasetProfile().view()
         self.profile_queue = asyncio.Queue()
         asyncio.create_task(self.read_profiles())
 
-    # TODO is this required? This is the only thing ensuring that `self.profile` isn't modififed concurrently. Should I treat these deployments as "controllers" in typical server frameworks, requiring thread local objects?
     async def read_profiles(self):
         while True:
             profile = await self.profile_queue.get()
             self.profile = self.profile.merge(profile)
 
     def log(self, df: pd.DataFrame):
         profile = ray.get(log_frame.remote(df))
+        # TODO is it better to just send the dataframes here via the queue? What costs more, tracking or merging?
         asyncio.create_task(self.profile_queue.put(profile))
 
     async def __call__(self, request: Request):
-        return str(self.profile.to_summary())
+        try:
```

Comment:
I could be sending dataframes between ray nodes instead of profiles but I'm not sure which is better. I assume it comes down to the size of the dataframe?

Reply:
profiles scale well in terms of staying small as you process potentially large dataframes.